crates/ra_lsp_server/src/vfs_filter.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::PathBuf;
+use ra_project_model::ProjectRoot;
+use ra_vfs::{RootEntry, Filter, RelativePath};
+
+pub struct IncludeRustFiles {
+    /// Is a member of the current workspace
+    is_member: bool,
+}

```

Comment:
Let's move this type to `project_model`, but make `include_dir` and `include_file` inherent methods. That way, batch and lsp_server would need to write only trivial impls, but the project_model would still be independent from vfs.

Reply:
Hmm, it would still require adding `relative_path` as dependency since the filter functions expects a `RelativePath`

Reply:
yeah, that should be ok

Reply:
relative path is pure (no io, no syscalls), so that's a light dependency